Components/WebService/Rest/Middleware.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Smartbox\Integration\FrameworkBundle\Components\WebService\Rest;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Smartbox\Integration\FrameworkBundle\Components\WebService\Rest\Exceptions\HttpClientRequestException;
+
+/**
+ * Guzzle Middlewae to be used by HttpClient.
+ */
+class Middleware
+{
+    public static $truncateResponseSize = 120;
+
+    /**
+     * This method override the default one used by Guzzle.
+     * Using this we can handle the RequestException the way we want by also overriding it with HttpClientRequestException.
+     *
+     * If you need to manage the size of response, set it as int at truncate_response_size config parameter
+     *
+     * @return \Closure
+     */
+    public static function httpErrors()
+    {
+        return function (callable $handler) {
+            return function (RequestInterface $request, array $options) use ($handler) {
+                return $handler($request, $options)->then(
+                    function (ResponseInterface $response) use ($request, $handler, $options) {
+                        $code = $response->getStatusCode();
+                        if ($code < 400) {
+                            return $response;
+                        }
+
+                        if (isset($options['truncate_response_size'])) {
+                            self::$truncateResponseSize = $options['truncate_response_size'];
+                        }
+
+                        throw HttpClientRequestException::create($request, $response);
+                    }
+                );
+            };
+        };
+    }
+
+    public static function handleResponseBody(ResponseInterface $response)
+    {
+        $body = $response->getBody();
+
+        if (!$body->isSeekable()) {
+            return null;
+        }
+
+        $size = $body->getSize();
+
+        if (0 === $size) {
+            return null;
+        }
+
+        $summary = $body->read($size);
+
+        if (self::$truncateResponseSize > 0 && $size > self::$truncateResponseSize) {
+            if (false === mb_detect_encoding($summary, 'UTF-8', true)) {
```

Comment:
Why do you want to validate if the content is in UTF-8? I don't like the idea of converting the response to something else... It might end up hiding the real error in the request. Plus `mb_substr` should be smart enough to truncate the string safely.

Reply:
Removed